src/app/pages/catalog/catalog.component.ts

```diff
@@ -1,105 +1,72 @@
 import { Component, OnDestroy, OnInit } from '@angular/core';
-import {
-  PageTitles,
-  ProductTypes,
-  ProductTypesType,
-} from '../../shared/constants/contstans';
-import { Subscription, switchMap, tap } from 'rxjs';
+import { PageTitles, ProductTypes } from '../../shared/constants/contstans';
+import { Observable, Subscription } from 'rxjs';
 import { ProductsService } from '../../shared/services/products.service';
-import { IProducts, StoreSelectors } from '../../shared/interfaces/interfaces';
-import { Store } from '@ngrx/store';
-
+import {
+  AppStateInterface,
+  IProducts,
+} from '../../shared/interfaces/interfaces';
+import { select, Store } from '@ngrx/store';
+import {
+  loadingSelector,
+  productsSelector,
+} from '../../store/products-store/products.selectors';
+import {
+  addProductsAction,
+  runLoadingAction,
+} from '../../store/products-store/products.actions';
+import { productTypeSelector } from '../../store/product-type-store/product-type.selectors';
 @Component({
   selector: 'app-catalog',
   templateUrl: './catalog.component.html',
 })
 export class CatalogComponent implements OnInit, OnDestroy {
   title?: string;
-  productList?: IProducts[];
-  loading: boolean = false;
+  productList?: Observable<IProducts[]>;
```

Comment:
I suggest to use 'productList$' variable instead. 
All async variables (Observables, Subjects, Promisess, etc... ) should have '$' in the end.
It will increase readability of the code.